File: app/src/main/java/seigneur/gauvain/mycourt/ui/shotDetail/ShotDetailActivityModule.java
package seigneur.gauvain.mycourt.ui.shotDetail;

import android.app.Activity;

import dagger.Binds;
import dagger.Module;
import seigneur.gauvain.mycourt.di.scope.PerActivity;

/**
 * Provides dependencies.
 */
@Module
public abstract class ShotDetailActivityModule {
    /**
     *
     * @param mShotDetailActivity
     * @return
     */
    @Binds
    @PerActivity
    abstract Activity activity(ShotDetailActivity mShotDetailActivity);



}

File: app/src/main/java/seigneur/gauvain/mycourt/ui/shotDetail/ShotDetailActivityModule.kt
package seigneur.gauvain.mycourt.ui.shotDetail

import android.app.Activity

import dagger.Binds
import dagger.Module
import seigneur.gauvain.mycourt.di.scope.PerActivity

/**
 * Provides dependencies.
 */
@Module
abstract class ShotDetailActivityModule {
    /**
     *
     * @param mShotDetailActivity
     * @return
     */
    @Binds
    @PerActivity
    internal abstract fun activity(mShotDetailActivity: ShotDetailActivity): Activity


}